CMakeLists.txt

```diff
@@ -35,9 +35,9 @@ option(WITH_ASAN "Enable address sanitizer" OFF)
 if(WITH_ASAN)
   message(STATUS "Enabling address sanitizer")
   add_compile_options(-fno-omit-frame-pointer -fno-common -fsanitize=address)
-  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -fsanitize=address -lasan" )
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address -lasan" )
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address -lasan" )
+  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -fsanitize=address -lasan -static-libasan" )
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address -lasan -static-libasan" )
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address -lasan -static-libasan" )
```

Comment:
Why has this additional flag become necessary?

Reply:
Not sure, I am getting weird error messages on Ubuntu otherwise (both with clang and gcc), regarding linking/preload order, or the wrong runtime `asan` being loaded. Might have to do with the dataset library being statically linked.

If you know of a better solution, I'd be happy to hear it.

Reply:
Makes sense if you were having linking problems.